scripts/artifacts-building/user_rcbops_artifacts_building.yml

```diff
@@ -31,3 +31,30 @@ openstack_repo_url: "{{ rpco_mirror_base_url }}"
 # default variant.
 lxc_image_cache_server: images.linuxcontainers.org
 
+# When we build python/container artifacts we pull the default
+# LXC cache variant from LXC upstream which may have packages
+# installed which are more recent than those we have available
+# in our artifacted apt repository. Here we figure out which
+# packages are installed, but not available in a configured
+# source, then we downgrade them to the latest available in
+# the configured sources.
+lxc_cache_prep_pre_commands: |
+    if [ -a /etc/resolv.conf ]; then
+      mv /etc/resolv.conf /etc/resolv.conf.org
+    fi
+    {% for resolver in lxc_cache_prep_dns %}
+    echo "nameserver {{ resolver }}" >> /etc/resolv.conf
+    {% endfor %}
+    apt-key add /root/repo.keys
+    export DEBIAN_FRONTEND=noninteractive
+    apt-get update
+    if apt list --installed 2>/dev/null | grep '\/now'; then
+      pkg_downgrade_list=$(apt list --installed 2>/dev/null | awk -F / '/\/now/ {print $1}')
+      pkg_downgrade_list_versioned=$(for pkg in ${pkg_downgrade_list}; do echo -n "${pkg}=$(apt-cache madison ${pkg} | head -n 1 | awk '{ print $3 }') "; done)
+      apt-get install -y --force-yes ${pkg_downgrade_list_versioned}
```

Comment:
By the way, the original state is as follows:

```
libc-bin/now 2.19-0ubuntu6.11 amd64 [installed,local]
libc6/now 2.19-0ubuntu6.11 amd64 [installed,local]
libgnutls-openssl27/now 2.12.23-12ubuntu2.7 amd64 [installed,local]
libgnutls26/now 2.12.23-12ubuntu2.7 amd64 [installed,local]
multiarch-support/now 2.19-0ubuntu6.11 amd64 [installed,local]
```

They're therefore all marked as explicitly installed anyway.